crates/oxc_linter/src/rules/import/no_named_as_default_member.rs

```diff
@@ -0,0 +1,166 @@
+#![allow(clippy::significant_drop_tightening)]
+use std::collections::HashMap;
+
+use dashmap::mapref::one::Ref;
+use oxc_ast::{
+    ast::{BindingPatternKind, Expression, MemberExpression},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{Atom, Span};
+use oxc_syntax::module_record::ImportImportName;
+
+use crate::{context::LintContext, rule::Rule};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-import(no-named-as-default-member): {1:?} also has a named export {2:?}")]
+#[diagnostic(severity(warning), help("Check if you meant to write `import {{{2:}}} from {3:?}`"))]
+struct NoNamedAsDefaultMemberDignostic(#[label] pub Span, String, String, String);
+
+/// <https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-named-as-default-member.md>
+#[derive(Debug, Default, Clone)]
+pub struct NoNamedAsDefaultMember;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Reports use of an exported name as a property on the default export.
+    ///
+    /// ### Example
+    ///
+    /// ```javascript
+    /// // ./bar.js
+    /// export function bar() { return null }
+    /// export default () => { return 1 }
+    ///
+    /// // ./foo.js
+    /// import bar from './bar'
+    /// const bar = foo.bar // trying to access named export via default
+    /// ```
+    NoNamedAsDefaultMember,
+    nursery
+);
+
+impl Rule for NoNamedAsDefaultMember {
+    fn run_once(&self, ctx: &LintContext<'_>) {
+        let module_record = ctx.semantic().module_record();
+
+        let mut has_members_map: HashMap<&Atom, (Ref<'_, Atom, _, _>, String)> = HashMap::default();
+        for import_entry in &module_record.import_entries {
+            let ImportImportName::Default(_) = import_entry.import_name else {
+                continue;
+            };
+
+            let specifier = import_entry.module_request.name();
+            let Some(remote_module_record_ref) = module_record.loaded_modules.get(specifier) else {
+                continue;
+            };
+
+            if !remote_module_record_ref.exported_bindings.is_empty() {
+                has_members_map.insert(
+                    import_entry.local_name.name(),
+                    (remote_module_record_ref, import_entry.module_request.name().to_string()),
+                );
+            }
+        }
+
+        if has_members_map.is_empty() {
+            return;
+        };
+        let get_external_module_name_if_has_entry = |module_name: &Atom, entry_name: &Atom| {
+            has_members_map.get(&module_name).and_then(|it| {
+                if it.0.exported_bindings.contains_key(entry_name) {
```

Comment:
Seem like will have a problem when encounter the same key in local scope.
```js
import foo from 'foo.js' 

{
   const foo = { bar: 'Hi'}
   console.log(foo.bar)
}
```

Reply:
Do you have an idea how to avoid it?

Reply:
What about finding import statement in `ctx.symbols().resolved_references`?

https://github.com/oxc-project/oxc/blob/0d77e1e788440ca4d5fe047ca4bdea7a01ed6210/crates/oxc_linter/src/utils/jest.rs#L198-L210

Reply:
I've added this test case to tests and there is no false positive

Reply:
I am a bit curious about why `foo.bar` not be visited when I print all the nodes in `ctx.semantic().nodes().iter()`. I'll check it after back to home.

Reply:
I thought it because of block, but I've added test that checks for error with block and it's working fine.
```rs
        r"import baz from './named-exports'; 
        { 
            const a = baz.a;
        }",
```